src/app/zap-templates/call-command-handler-src.zapt

```diff
@@ -100,15 +100,64 @@ EmberAfStatus emberAf{{asCamelCased name false}}Cluster{{asCamelCased side false
         {{#if (isCommandAvailable parent.side incoming outgoing commandSource name)}}
         case ZCL_{{asDelimitedMacro name}}_COMMAND_ID: {
             {{#if (zcl_command_arguments_count this.id)}}
-            uint32_t argOffset = 0;
+            uint32_t payloadOffset = cmd->payloadStartIndex;
             {{#zcl_command_arguments}}
-            {{asUnderlyingType type}} * {{asSymbol label}} = ({{asUnderlyingType type}} *)(cmd->buffer + argOffset);
+            {{#if (isNullable type isArray)}}
+            uint8_t * {{asSymbol label}};
+            {{else if (isStrEqual label "groupId")}}
+            GroupId {{asSymbol label}};
+            {{else}}
+            {{asUnderlyingType type}} {{asSymbol label}};
+            {{/if}}
+            {{/zcl_command_arguments}}
+
+            {{#zcl_command_arguments}}
+            {{#if presentIf}}
+            if ({{presentIf}})
+            {
+            {{/if}}
+            {{#if isArray}}
+            {{asSymbol label}} = (uint8_t *)(cmd->buffer + payloadOffset);
+            {{else if (isString type)}}
+            {{asSymbol label}} = emberAfGetString(cmd->buffer, payloadOffset, cmd->bufLen);
+            {{else if (isLongString type)}}
+            {{asSymbol label}} = emberAfGetLongString(cmd->buffer, payloadOffset, cmd->bufLen);
+            {{else if (isStrEqual label "groupId")}}
+            {{asSymbol label}} = (*(GroupId *)(cmd->buffer + payloadOffset));
+            {{else}}
+            {{asSymbol label}} = (*({{asUnderlyingType type}} *)(cmd->buffer + payloadOffset));
+            {{/if}}
             {{#unless (isLastElement index count)}}
-            argOffset+= sizeof({{asUnderlyingType type}});
+            {{#if (isString type)}}
+            payloadOffset += emberAfStringLength({{asSymbol label}}) + 1u;
+            {{else if (isLongString type)}}
+            payloadOffset += emberAfLongStringLength({{asSymbol label}}) + 2u;
+            {{else}}
+            payloadOffset += sizeof({{asUnderlyingType type}});
+            {{/if}}
             {{/unless}}
+            {{#if presentIf}}
+            }
+            else
+            {
+            {{#if (isNullable type isArray)}}
+            {{asSymbol label}} = NULL;
+            {{else if (isStrEqual type "INT8U")}}
+            {{asSymbol label}} = UINT8_MAX;
+            {{else if (isStrEqual type "INT16U")}}
+            {{asSymbol label}} = UINT16_MAX;
+            {{else if (isStrEqual type "INT32U")}}
+            {{asSymbol label}} = UINT32_MAX;
+            {{else if (isStrEqual type "INT65U")}}
+            {{asSymbol label}} = UINT64_MAX;
+            {{else if (isStrEqual type "CLUSTER_ID")}}
+            {{asSymbol label}} = EMBER_AF_INVALID_CLUSTER_ID
+            {{/if}}
```

Comment:
Can we have an "else" here that errors out or something?  Otherwise seems like we would end up working with uninitialized values.

Reply:
> Can we have an "else" here that errors out or something? Otherwise seems like we would end up working with uninitialized values.

Yes. I can add that. 
fwiw I looked into the definition of the `Silabs` XML and look at all the cases where it could happen. But I agree that having an else case that log something will at least tell us when the ZCL spec has changed and we have have not updated this case.
But in general that is the same issue than previously. I can not really write a case per type, so it's not easy to write it as I wish.

Reply:
I end up generating a comment. If a case is not handled, the compiler will complains with potentially uninitialised value for the developer and on CI.

Reply:
@vivien-apple All of this should go into a CHIP specific helper in order to clean the template. the helper would be something like `{{ chipDefaultInvalidValue type isArray}}` that way we could shrink all of this to `{{asSymbol label}} = {{ chipDefaultInvalidValue type isArray}}`

Reply:
The main issue I have is the fact that I can't access the underlying type of the data. So it's hard for me to do a generic helper that will work in all cases.

Do you know if there is a way to pass `{{asUnderlyingType}}` to a helper ? So far the runtime got stuck when I'm running such things.